src/mirth/elab.mth

```diff
@@ -981,14 +987,16 @@ def(data-word-new!, Data Str Int -- Word,
 
 def(elab-data-done!, Data --,
     \(dat ->
-        dat is-enum? then(
+        dat is-resource? not then(
             dat "tag" 0 data-word-new! \(tag ->
                 CTX0 dat TData T1 TYPE_INT T1 T-> ready2 tag ~ctx-type !
```

Comment:
the input type and context `CTX0 dat TData T1` are not quite right, because the data type could have type arguments (unlike enums which are monomorphic). Actually computing the input type might be kind of a hassle though. You can see how it's done in elab-data-tag. As a shortcut, maybe you can grab one of the tags (there's guaranteed to be at least one) and use its output type and context for this.